### PR TITLE
doc: add xgboost documentation for inference

### PR DESCRIPTION
**Pull request by edwardjkim**
*Issue #, if available:*

*Description of changes:*

Adding XGBoost documentation for inference.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/frameworks/xgboost/using_xgboost.rst [hunk 1]
```diff
@@ -179,24 +179,228 @@ After you create an estimator, call the ``fit`` method to run the training job.
...
+
+-  The first argument is expected to be the return value from input_fn.
+-  The second argument is the loaded model.
+-  The return value should be of the correct type to be passed as the first argument to ``output_fn``.
+   If you use the default ``output_fn``, this should be a NumPy array.
+
+Process Output
+""""""""""""""
+
+After invoking ``predict_fn``, the model server invokes ``output_fn``, passing in the return-value from
```
Comment (metrizable) [inline doc/frameworks/xgboost/using_xgboost.rst:379]: remove hyphen?

```suggestion
After invoking ``predict_fn``, the model server invokes ``output_fn``, passing in the return value from
```

File: doc/frameworks/xgboost/using_xgboost.rst [hunk 2]
```diff
@@ -179,24 +179,228 @@ After you create an estimator, call the ``fit`` method to run the training job.
...
+Where ``input_object`` is the object returned from ``input_fn`` and ``model`` is the model loaded by ``model_fn``.
+
+The default implementation of ``predict_fn`` invokes the loaded model's ``predict`` function on ``input_object``,
+and returns the resulting value. The return-type should be a NumPy array to be compatible with the default
+``output_fn``.
+
+The example below shows an overriden ``predict_fn`` that returns a two-dimensional NumPy array where
+the first columns are predictions and the remaining columns are the feature contributions
+(`SHAP values <https://github.com/slundberg/shap>`_) for that prediction.
+When ``pred_contribs`` is True in ``xgboost.Booster.predict()``, the output will be a matrix of size
```
Comment (metrizable) [inline doc/frameworks/xgboost/using_xgboost.rst:355]: typeset suggestion:

```suggestion
When ``pred_contribs`` is ``True`` in ``xgboost.Booster.predict()``, the output will be a matrix of size
```

File: doc/frameworks/xgboost/using_xgboost.rst [hunk 3]
```diff
@@ -179,24 +179,228 @@ After you create an estimator, call the ``fit`` method to run the training job.
...
+
+where ``request_body`` is a byte buffer and ``request_content_type`` is a Python string.
+
+The SageMaker XGBoost model server provides a default implementation of ``input_fn``.
+This function deserializes CSV, LIBSVM, or protobuf recordIO into a ``xgboost.DMatrix``.
+
+Default csv deserialization requires ``request_body`` contain one or more lines of CSV numerical data.
+The data is first loaded into a two-dimensional array, where each line break defines the boundaries of the first
+dimension, and then it is converted to an `xgboost.Dmatrix`.
+It assumes that CSV input does not have the label column.
```
Comment (metrizable) [inline doc/frameworks/xgboost/using_xgboost.rst:310]: `@ignorable` nit-style: bring to previous line?

```suggestion
dimension, and then it is converted to an `xgboost.Dmatrix`. It assumes that CSV input does not have the
label column.
```

File: doc/frameworks/xgboost/using_xgboost.rst [hunk 4]
```diff
@@ -179,24 +179,228 @@ After you create an estimator, call the ``fit`` method to run the training job.
...
+When a request is made against an endpoint running a SageMaker XGBoost model server, the model server receives two
+pieces of information:
+
+-  The request Content-Type, for example "application/x-npy" or "text/libsvm"
+-  The request data body, a byte array
+
+The SageMaker XGBoost model server will invoke an ``input_fn`` function in your inference script, passing in this
+information.
+If you define an ``input_fn`` function definition, it should return an object that can be passed to ``predict_fn`` and
+have the following signature:
```
Comment (metrizable) [inline doc/frameworks/xgboost/using_xgboost.rst:296]: `@ignorable` nit-style: bring to previous line?

```suggestion
information. If you define an ``input_fn`` function definition, it should return an object that can be passed
to ``predict_fn`` and have the following signature:
```

**Comment by edwardjkim** [comment]
Sorry about that. I just squashed the commits so there should no change in contents.

**Pull request merged**